test/sqllogictest/cluster_log_drop.slt

```diff
@@ -21,7 +21,7 @@ mode cockroach
 query T rowsort
 SELECT (SELECT COUNT(*) FROM mz_catalog.mz_sources WHERE name NOT LIKE '%_1') - (SELECT COUNT(*) FROM mz_catalog.mz_sources WHERE name LIKE '%_1');
 ----
-0
+1
```

Comment:
This LGTM. One alternative thing you could do to make this less annoying when adding more storage collections in the future is filter on `type = 'log'`.